src/services/bonusService.ts

```diff
@@ -51,8 +62,50 @@ export default class BonusService {
             return ResponseErrorConflict(readableProblem(response.value));
           case 500:
             return ResponseErrorInternal(readableProblem(response.value));
+          case 401:
+            // should not happen, thus is an error
```

Comment:
401 errors are handled by the passport middleware that checks the autentication, you should not handle it directly in the controller / service

Reply:
the 401 returned here is from the underlying functions api, thus it's not the one handled by `passport`. Is an apikey-based auth, so it might not happen but for misconfiguration (am I right?). That's why I handled it as an error rather than an auth failure.

The point of handling this explicitly rather than as a `default` case is to make eloquent the difference between statuses we choose not to handle (like this) and statuses the functions api didn't define in the specs, but it's actually returning at runtime. 
Also, this way I could implement an exhaustive check on client's statuses, and this may help in future api changes.

Reply:
@gunzip the question is: should we trace such cases with special logging?

Reply:
errors are automatically traced by application insights